hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis_snapshot/OmRatisSnapshotProvider.java

```diff
@@ -76,7 +81,7 @@ public class OmRatisSnapshotProvider extends RDBSnapshotProvider {
   private final Map<String, OMNodeDetails> peerNodesMap;
   private final HttpConfig.Policy httpPolicy;
   private final boolean spnegoEnabled;
-  private final URLConnectionFactory connectionFactory;
+  private URLConnectionFactory connectionFactory;
```

Comment:
When possible, I prefer not to remove the final modifier just to make tests work.  Could we change the setConnectionFactory() to a getConnectionFactory() method, and mock that method in the test?

Reply:
Unfortunately I could not find any other way. I need to mock `connectionFactory` field which is not a part of constructor parameters. I tried using `@Mock`/`@IncjectMock` combination, `Reflection` and `Unsafe`, but without results. Another idea would be to create constructor specificly for testing purposes, but I do not like that approach.

Do you have an idea how this could be handled?

Reply:
I just meant something like this:  https://github.com/GeorgeJahad/ozone/compare/da4e99768a..urlConnectionFactory

Reply:
Oh, that is totally a different story. Sure thing!

Reply:
> When possible, I prefer not to remove the final modifier just to make tests work. Could we change the setConnectionFactory() to a getConnectionFactory() method, and mock that method in the test?

+1 on using final.

Don't have very strong opinion but I think it would be nicer to pass `URLConnectionFactory` as constructor parameter to achieve deterministic behavior for testing. Also aligns with Dependency injection.

Reply:
Thanks @hemantk-12 for the review! I added constructor which could be used for testing but also alligns with DI.